great_expectations/data_context/store/ge_cloud_store_backend.py

```diff
@@ -366,3 +367,9 @@ def _has_key(self, key: Tuple[str, ...]) -> bool:
     @property
     def config(self) -> dict:
         return self._config
+
+    def build_key(self, id_: Optional[str] = None, **kwargs) -> GeCloudIdentifier:
```

Comment:
Moved the key generation which is specific to the store backend type to the store backend.

Reply:
oh wow this is a really good call